src/ReverseProxy/Configuration/IConfigNotifier.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Yarp.ReverseProxy.Configuration;
+
+/// <summary>
+/// Allows configuration management to optionaly notify of errors occuring during loading and applying.
+/// In order to leverage this service, one has got to register a type implementing this interface in
+/// the dependency injection container.
+/// </summary>
+public interface IConfigNotifier
```

Comment:
```suggestion
public interface IConfigChangeNotifier
```

@Tratcher thoughts on the name here?

Reply:
Yeah, that's a little clearer.

Reply:
Fixed